File: golang/internal/infrastructure/server/dummy_server.go
package server

import (
	assetusecase "folio/codinginterview/internal/application/usecase/asset"
	"folio/codinginterview/internal/application/usecase/order"
	portfoliousecase "folio/codinginterview/internal/application/usecase/portfolio"
	infrarepo "folio/codinginterview/internal/infrastructure/repository"
	"folio/codinginterview/internal/presentation"
)

type DummyServer struct {
	AssetController     *presentation.AssetController
	PortfolioController *presentation.PortfolioController
	OrderController     *presentation.OrderController
}

func NewDefaultDummyServer() *DummyServer {
	portfolioRepo := infrarepo.NewPortfolioRepositoryImpl()
	accountRepo := infrarepo.NewAccountRepositoryImpl()

	getAssetUsecase := assetusecase.NewGetAssetUsecase(accountRepo)
	getLatestPortfolioUsecase := portfoliousecase.NewGetLatestPortfolioUsecase(portfolioRepo)
	updatePortfolioUsecase := portfoliousecase.NewUpdatePortfolioUsecase(portfolioRepo)
	newOrderUsecase := order.NewNewOrderUsecase(accountRepo, portfolioRepo)
	additionalBuyOrderUsecase := order.NewAdditionalBuyOrderUsecase(accountRepo, portfolioRepo)
	rebalanceOrderUsecase := order.NewRebalanceOrderUsecase(accountRepo, portfolioRepo)

	assetController := presentation.NewAssetController(getAssetUsecase)
	portfolioController := presentation.NewPortfolioController(getLatestPortfolioUsecase, updatePortfolioUsecase)
	orderController := presentation.NewOrderController(newOrderUsecase, additionalBuyOrderUsecase, rebalanceOrderUsecase)

	return &DummyServer{
		AssetController:     assetController,
		PortfolioController: portfolioController,
		OrderController:     orderController,
	}
}

File: typescript/src/infrastructure/server/dummyServer.ts
import { GetAssetUsecase } from "../../application/usecase/asset/getAssetUsecase";
import { AdditionalBuyOrderUsecase } from "../../application/usecase/order/additionalBuyOrderUsecase";
import { NewOrderUsecase } from "../../application/usecase/order/newOrderUsecase";
import { RebalanceOrderUsecase } from "../../application/usecase/order/rebalanceOrderUsecase";
import { GetLatestPortfolioUsecase } from "../../application/usecase/portfolio/getLatestPortfolioUsecase";
import { UpdatePortfolioUsecase } from "../../application/usecase/portfolio/updatePortfolioUsecase";
import { AssetController } from "../../presentation/assetController";
import { OrderController } from "../../presentation/orderController";
import { PortfolioController } from "../../presentation/portfolioController";
import { AccountRepositoryImpl } from "../repository/accountRepositoryImpl";
import { PortfolioRepositoryImpl } from "../repository/portfolioRepositoryImpl";

export class DummyServer {
  readonly assetController: AssetController;
  readonly portfolioController: PortfolioController;
  readonly orderController: OrderController;

  constructor(
    assetController: AssetController,
    portfolioController: PortfolioController,
    orderController: OrderController,
  ) {
    this.assetController = assetController;
    this.portfolioController = portfolioController;
    this.orderController = orderController;
  }

  static default(): DummyServer {
    const portfolioRepository = new PortfolioRepositoryImpl();
    const accountRepository = new AccountRepositoryImpl();

    const getAssetUsecase = new GetAssetUsecase(accountRepository);
    const getLatestPortfolioUsecase = new GetLatestPortfolioUsecase(portfolioRepository);
    const updatePortfolioUsecase = new UpdatePortfolioUsecase(portfolioRepository);
    const newOrderUsecase = new NewOrderUsecase(accountRepository, portfolioRepository);
    const additionalBuyOrderUsecase = new AdditionalBuyOrderUsecase(
      accountRepository,
      portfolioRepository,
    );
    const rebalanceOrderUsecase = new RebalanceOrderUsecase(accountRepository, portfolioRepository);

    const assetController = new AssetController(getAssetUsecase);
    const portfolioController = new PortfolioController(
      getLatestPortfolioUsecase,
      updatePortfolioUsecase,
    );
    const orderController = new OrderController(
      newOrderUsecase,
      additionalBuyOrderUsecase,
      rebalanceOrderUsecase,
    );

    return new DummyServer(assetController, portfolioController, orderController);
  }
}
